syntax = "proto3";

package clueless;

service Clueless{
  rpc ServerGreetings(Greetings) returns (GreetingsResponse) {}
  rpc Events (EventsRequest) returns (stream Event) {}

}

// Client requests

message Greetings{
  string client_name = 1;
}

message EventsRequest {
  bool request = 1;
}

// Server responses

message GreetingsResponse {
  string message = 1;
  string status = 2;
}

message Event {
  string status = 1;
  repeated string players = 2;
}